### DISCUSSION TITLE
Scaling down Conductor server

### DISCUSSION THREAD

**Discussion by muthurajr** [Netflix/conductor#2404] [Q&A]
Hi,

I'm noticing a scenario where the Lambda tasks are not rescheduled when the server instance is killed.

The workflow executed contains various system Lambda tasks and are executed in parallel. When the conductor server is scaled up, no issues and can notice more tasks are being executed in parallel. But once conductor server scales down due to usage, the system Lambda tasks which were running in the killed servers are stuck in "RUNNING" status. The tasks are not rescheduled by the running server instance. Potential problems which I can think of are

1. Workflow locked by an instance is not picked up by another instance
2. Response Timeout settings are not configured for the system tasks

**Environment:**
```
Conductor version: 3.0.6
Infrastructure: Kubernetes
Conductor Server: 1 pod with requests(1 cpu/1GB memory) & limits (2 cpu/2GB memory), and scaling enabled with 1 to 5 based on cpu utilization
Workflow Lock: Redis
Persistence Store: Postgres
```

Any help is much appreciated to find the cause the issue!

**Comment by apanicker-nflx** [comment]
Currently, LAMBDA tasks are executed inline by the Conductor server synchronously in the workflow evaluation path. If the instance were to be terminated abruptly when this task is being executed, it is possible that this synchronous evaluation is not replayed. We are considering different approaches to improve resiliency and auto-recovery in such scenarios.

>But once conductor server scales down due to usage

However, this indicates that the server instance termination is not abrupt, rather a graceful shutdown performed as part of the downscaling process. Can you share what type of expressions you are running as part of your LAMBDA tasks? Have you considered converting these to SIMPLE tasks so that the server does not bear the load of executions?

**Reply by muthurajr** [reply]
@apanicker-nflx Thank you so much for your response!

The actual production workflows we implemented are using the SIMPLE tasks. My intention here was to achieve scaling in conductor server. I was trying to create a heavy workload using lambda tasks(just sleep for 10ms) just for testing purpose.

Two additional questions if you have sometime to answer them.

1. Is auto-recovery the problem with lambda tasks only or with other system tasks also? An example can be Kafka Publish or HTTP task which is being killed by the server shutdown.
2. Do we have any parameter in the metrics to consider for scaling the conductor server?
